### PR TITLE
[bug] Fix prolongation [pi-prolong]

### PR DESCRIPTION
**Pull request by camierjs**
- Add the input prolongation in `InitTVectors` to be able to test it
- Rename few unit tests under `linalg`
- Add `test_SquareMatrix` that would have shown the issue

 | PR | Author | Editor | Reviewers | Assignment | Approval | Merge| 
 | --- | --- | --- | --- | --- | --- | --- | 
| [#1320](https://github.com/mfem/mfem/pull/1320) | @camierjs | @tzanio | @v-dobrev + @tzanio | 02/24/20 | 02/25/20 | 02/27/20 | |

### CODE REVIEW

**Comment by tzanio** [comment]
Why the renaming? The tests so far seem to consistently use lowercase after `test_`.

**Comment by v-dobrev** [comment]
> Why the renaming? The tests so far seem to consistently use lowercase after `test_`.

It looks like there was already some capitalization, e.g. `test_blockMatrix.cpp` but you are right -- most were lowercase.

One advantage of using lowercase names is that it is easier to type in a terminal 😄 and the other is consistency with the source filenames. Either way, I do not have a strong preference for this directory.

File: tests/unit/linalg/test_matrix_square.cpp [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+
+            x0 = 0.0;
+            b = 1.0;
+            pa.SetAssemblyLevel(AssemblyLevel::PARTIAL);
+            pa.AddDomainIntegrator(new DiffusionIntegrator(one));
+            pa.Assemble();
+            pa.FormLinearSystem(ess_tdof_list, x0, b, A_pa, X[0], B[0]);
+            Solver *M_pa = new OperatorJacobiSmoother(pa, ess_tdof_list);
+            CGSolver cg_pa(MPI_COMM_WORLD);
+            cg_pa.SetRelTol(EPS*EPS);
```
Comment (v-dobrev) [inline tests/unit/linalg/test_matrix_square.cpp]: Unlike the `PCG` function, the tolerance here should not be squared: `EPS*EPS` -> `EPS`.

File: tests/unit/linalg/test_matrix_square.cpp [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+            x1 = 0.0;
+            b = 1.0;
+            fa.SetAssemblyLevel(AssemblyLevel::FULL);
+            fa.AddDomainIntegrator(new DiffusionIntegrator(one));
+            fa.Assemble();
+            fa.FormLinearSystem(ess_tdof_list, x1, b, A_fa, X[1], B[1]);
+            HypreBoomerAMG *M_fa = new HypreBoomerAMG();
+            CGSolver cg_fa(MPI_COMM_WORLD);
+            cg_fa.SetRelTol(EPS*EPS);
```
Comment (v-dobrev) [inline tests/unit/linalg/test_matrix_square.cpp]: Same as above: `EPS*EPS` -> `EPS`.

File: tests/unit/linalg/test_matrix_square.cpp (hunk 1 above)
Comment (camierjs) [reply tests/unit/linalg/test_matrix_square.cpp]: `done`

File: tests/unit/linalg/test_matrix_square.cpp (hunk 2 above)
Comment (camierjs) [reply tests/unit/linalg/test_matrix_square.cpp]: `done`

**Comment by v-dobrev** [comment]
Merged in `next` for testing...

**Comment by v-dobrev** [comment]
Oops! @tzanio I thought you already approved this.

**Pull request merged**